### DISCUSSION TITLE
Does FileStream(SafeFileHandle) have a independent Position (with .NET 6)

### DISCUSSION THREAD

**Discussion by msedi** [dotnet/runtime#64209] [Q&A]
Originally since the `FileStream` was working with `Set/GetFilePointer` the problem was that the use of a `FileStream `allocated with 

```cs
FileStream originalStream;
FileStream newStream = new FileStream(originalStream.SafeFileHandle, FileAccess.Read);

```

was not threadsafe. Since you changed that `FileStream` has now its internal FilePointer without accessing the `Set/GetFilterPointer` it seems to me that this is threadsafe now, but couldn't find any note on this. **Can someone please clarify?**

**Comment by stephentoub** [comment]
A FileStream instance should not be used concurrently.  If you need concurrent access, you can use the RandomAccess type introduced in .NET 6.

**Reply by msedi** [reply]
@stephentoub. Thx. I understand that the instance itself is not threadsafe. But originally (before the .NET 6 changes on the filestream) the problem was that even when creating two instances like above where I used originalStream in Thread 1 and newStream in Thread 2 the file pointer was shared. Is this something that shouldn't be done in general, if yes is then the use of the constructor only to hand over a handle created by `CreateFile`?

**Reply by msedi** [reply]
@stephentoub: Do you have a hint for me in this case. I'm not sure if I can use it as needed or if it currently simply works but crashes on some unknown condition?
